tests/catch/include/hip_test_context.hh

```diff
@@ -36,9 +36,15 @@ THE SOFTWARE.
 #if defined(_WIN32)
 #define HT_WIN 1
 #define HT_LINUX 0
+#define HT_MACOS 0
 #elif defined(__linux__)
 #define HT_WIN 0
 #define HT_LINUX 1
+#define HT_MACOS 0
+#elif defined(__APPLE__) || defined(__MACOSX)
+#define HT_WIN 0
+#define HT_LINUX 0
+#define HT_MACOS 1
```

Comment:
I have no idea where these new flags are used.
The existing code quality here is pretty horrible.